src/blocked.c

```diff
@@ -114,6 +114,7 @@ void updateStatsOnUnblock(client *c, long blocked_us, long reply_us, int had_err
         updateCommandLatencyHistogram(&(c->lastcmd->latency_histogram), total_cmd_duration*1000);
     /* Log the command into the Slow log if needed. */
     slowlogPushCurrentCommand(c, c->lastcmd, total_cmd_duration);
+    c->duration = 0;
```

Comment:
Could there be cases when this function is not called and duration is not 0-ed out causing in assertion failures?
For example looking at the code here, not all clauses call updateStatsOnUnblock:
https://github.com/redis/redis/blob/d0da0a6a3f2a599720be0e51272ef831b07b9dd2/src/blocked.c#L228

Reply:
actually I agree. we should make sure the wait timeout also calls updateStatsOnUnblock, it is a bug that wait does not report it.

Reply:
I don't follow what we are calling updateStatsOnUnblock for the `WAIT` command. The command is not being reprocessed like the other blocking commands, and it's done centrally inside another function?

Reply:
Synced offline. It seems like the thinking was that on blocking, we would always eventually handle the duration part. So, new fix for the wait command. @ranshid please confirm.